### PR TITLE
SA-1998: Added Form tracking component to industry benchmark modal

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed

- Added `<Form />` component to industry benchmark modal

### How To Test

- Add Inbox Folder Rate as metric in report builder
- Click Industry Benchmark above chart
- Verify that the industry benchmark form still works

### To Do

- [ ] Address feedback

### CODE REVIEW

**Review by nstanard — COMMENTED** [review]
image
Modal looks good.


image
Wondering how we are going to identify the selected option in the form submit. I'm not seeing the selected attribute on the option I picked. Testing more to see if I am missing something.

**Comment by nstanard** [comment]
> Wondering how we are going to identify the selected option in the form submit. I'm not seeing the selected attribute on the option I picked. Testing more to see if I am missing something.

image

It seems that if we need to have the <option /> that's selected indicated in the form submit we need Matchbox to set the selected attribute. I'm not seeing that happen right now.

**Comment by PatrickSison** [comment]
@nstanard I think that would be an issue with all forms now then right? We're doing most forms as having inputs externally controlled in react state, so not sure if those values would ever get passed through the submit handler.

**Review by nstanard — APPROVED** [review]
Alright - I got the info I needed....

We can look at the el.nodeName and as @nicklemmon helped point out in Slack we can look at the el.value for select dropdowns to get the value of the option they picked. Obviously that's not important for this PR but will come into play over the next few days/weeks.

image

LGTM!

File: src/pages/reportBuilder/components/IndustryBenchmarkModal.js [hunk 1]
```diff
@@ -38,11 +41,19 @@ export const IndustryBenchmarkModal = ({ metrics, isModalOpen, closeModal }) =>
...
+    setIndustryBenchmark(state);
+    closeModal();
+  };
+
   return (
-    <>
-      <Modal open={isModalOpen} onClose={closeModal} showCloseButton>
-        <Modal.Header>Industry Benchmark</Modal.Header>
-        <Modal.Content>
+    <Modal open={isModalOpen} onClose={closeModal} showCloseButton>
```
Comment (nstanard) [inline src/pages/reportBuilder/components/IndustryBenchmarkModal.js:53]: I could be wrong but at a glance I think we may also need a title attribute for the modals if this PR doesn't already handle it - https://github.com/SparkPost/2web2ui/pull/2112/files

Comment (nstanard) [reply src/pages/reportBuilder/components/IndustryBenchmarkModal.js:53]: Nvm - only legacy instances need title 🥇

**Pull request merged**